core/src/main/java/com/scalar/db/api/AuthAdmin.java

```diff
@@ -404,48 +404,68 @@ interface User {
     boolean isSuperuser();
   }
 
-  /** Represents a role. */
+  /** Represents a role, including its granted member roles. */
   interface Role {
+    /**
+     * Returns the role name.
+     *
+     * @return the role name
+     */
     String getName();
-  }
 
-  /** Represents a role with its hierarchy information. */
-  interface RoleDetail {
-    Role getRole();
-
-    List<RoleHierarchy> getRoleHierarchies();
+    /**
+     * Returns the member roles granted to the role.
+     *
+     * @return the member roles granted to the role
+     */
+    List<MemberRole> getMemberRoles();
```

Comment:
If the Role is managed in a tree structure, `getChildRoles` is also an option.

The term `child roles` seems to be used in some places at the first glance:
- https://docs.cloud.google.com/spanner/docs/fgac-about
- https://supabase.com/docs/guides/database/postgres/roles
- https://docs.oracle.com/en/cloud/saas/public-sector-compliance-regulation-common/21c/pssci/managing-roles.html#Managing-Roles-in-Public-Sector-Compliance-and-Regulation
- https://www.ibm.com/docs/en/sim/7.0.2?topic=roles-add-child